spyder/plugins/tests/test_init.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Copyright © Spyder Project Contributors
+#
+# Licensed under the terms of the MIT License
+# (see spyder/__init__.py for details)
+# -----------------------------------------------------------------------------
+
+"""Tests for the base plugin classes ('__init__.py')."""
+
+# Standard library imports
+import os
+try:
+    from unittest.mock import MagicMock
+except ImportError:
+    from mock import MagicMock  # Python 2
+
+# 3rd party imports
+import pytest
+from flaky import flaky
+from qtpy.QtCore import QEvent
+from qtpy.QtWidgets import QTabBar
+
+# Local imports
+from spyder.plugins import TabFilter
+from spyder.app import start
+from spyder.config.main import CONF
+
+
+# =============================================================================
+# Fixtures
+# =============================================================================
+@pytest.fixture
+def main_window(request):
+    """Main Window fixture"""
+    # Disable unneeded introspection to save on startup time
+    try:
+        os.environ.pop('SPY_TEST_USE_INTROSPECTION')
+    except KeyError:
+        pass
+
+    # Make sure single instance mode is disabled
+    CONF.set('main', 'single_instance', False)
+
+    # Start the window
+    window = start.main()
+
+    # Teardown
+    def close_window():
+        window.close()
+    request.addfinalizer(close_window)
+
+    return window
```

Comment:
Please don't copy this fixture here. Instead remove it and move `test_tabfilter_typeerror_full` to `test_mainwindow.py`.

Reply:
Thanks; makes sense. 'Tis done.